plugins/kubernetes-common/src/permissions.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  createPermission,
+  ResourcePermission,
+  BasicPermission,
+} from '@backstage/plugin-permission-common';
+
+/**
+ * Permission resource type which corresponds to cluster entities.
+ * @alpha
+ */
+export const RESOURCE_TYPE_KUBERNETES_RESOURCE = 'kubernetes-resource';
+
+/**
+ * Convenience type for kubernetes resource
+ * {@link @backstage/plugin-permission-common#ResourcePermission}s.
+ * @alpha
+ */
+export type KubernetesResourcePermission = ResourcePermission<
+  typeof RESOURCE_TYPE_KUBERNETES_RESOURCE
+>;
+
+/** This permission is used to authorize actions that involve one or more
+ * kubernetes resources.
+ * @alpha
+ */
+export const kubernetesClusterReadPermission: BasicPermission =
+  createPermission({
+    name: 'kubernetes.clusters',
+    attributes: {
+      action: 'read',
+    },
```

Comment:
I wonder if this should be a RESOURCE permission instead. How likely is it to implement fine grained permissions on different clusters?

Reply:
There may be a case to made for this -- some scenarios I can dream up:
* What if the catalog clusterlocator is in use? Maybe the clusters endpoint should only surface those clusters owned by the current user.
* If, say, the GKE clusterlocator is in use, perhaps integrators would want the clusters endpoint to only show clusters from a certain region, or clusters with certain metadata fields/tags
* Maybe a bit far-fetched, but maybe integrators only want the clusters endpoint to surface clusters with a particular authProvider (say, "only show clusters with server-side auth" for "business" type personas who don't have cloud accounts or "only show clusters with client-side auth" for more "dev" type personas who have fine-grained access to k8s objects)(